src/compositions/useLink.ts

```diff
@@ -61,112 +61,111 @@ export type UseLinkOptions = RouterPushOptions & {
 }
 
 type UseLinkArgs<
-  TSource extends RegisteredRoutesName,
-  TParams = RouteParamsByKey<RegisteredRoutes, TSource>
+  TRouter extends Router,
+  TSource extends RouterRouteName<TRouter>,
+  TParams = RouteParamsByKey<RouterRoutes<TRouter>, TSource>
 > = AllPropertiesAreOptional<TParams> extends true
   ? [params?: MaybeRefOrGetter<TParams>, options?: MaybeRefOrGetter<UseLinkOptions>]
   : [params: MaybeRefOrGetter<TParams>, options?: MaybeRefOrGetter<UseLinkOptions>]
 
-/**
- * A composition to export much of the functionality that drives RouterLink component. Can be given route details to discover resolved URL,
- * or resolved URL to discover route details. Also exports some useful context about routes relationship to current URL and convenience methods
- * for navigating.
- *
- * @param source - The name of the route or a valid URL.
- * @param params - If providing route name, this argument will expect corresponding params.
- * @param options - {@link RouterResolveOptions} Same options as router resolve.
- * @returns {UseLink} Reactive context values for as well as navigation methods.
- * @group Compositions
- */
-export function useLink<TRouteKey extends RegisteredRoutesName>(name: MaybeRefOrGetter<TRouteKey>, ...args: UseLinkArgs<TRouteKey>): UseLink
-export function useLink(url: MaybeRefOrGetter<Url>, options?: MaybeRefOrGetter<UseLinkOptions>): UseLink
-export function useLink(resolvedRoute: MaybeRefOrGetter<ResolvedRoute | undefined>, options?: MaybeRefOrGetter<UseLinkOptions>): UseLink
-export function useLink(source: MaybeRefOrGetter<string | ResolvedRoute | undefined>, paramsOrOptions?: MaybeRefOrGetter<Record<PropertyKey, unknown> | UseLinkOptions>, maybeOptions?: MaybeRefOrGetter<UseLinkOptions>): UseLink
-export function useLink(
-  source: MaybeRefOrGetter<string | ResolvedRoute | undefined>,
-  paramsOrOptions: MaybeRefOrGetter<Record<PropertyKey, unknown> | UseLinkOptions> = {},
-  maybeOptions: MaybeRefOrGetter<UseLinkOptions> = {},
-): UseLink {
-  const router = useRouter()
-
-  const route = computed(() => {
-    const sourceValue = toValue(source)
-    if (typeof sourceValue !== 'string') {
-      return sourceValue
-    }
+type UseLinkFunction<TRouter extends Router> = {
+  <TRouteKey extends RouterRouteName<TRouter>>(name: MaybeRefOrGetter<TRouteKey>, ...args: UseLinkArgs<TRouter, TRouteKey>): UseLink,
+  (url: MaybeRefOrGetter<Url>, options?: MaybeRefOrGetter<UseLinkOptions>): UseLink,
+  (resolvedRoute: MaybeRefOrGetter<ResolvedRoute | undefined>, options?: MaybeRefOrGetter<UseLinkOptions>): UseLink,
+  (source: MaybeRefOrGetter<string | ResolvedRoute | undefined>, paramsOrOptions?: MaybeRefOrGetter<Record<PropertyKey, unknown> | UseLinkOptions>, maybeOptions?: MaybeRefOrGetter<UseLinkOptions>): UseLink,
```

Comment:
This overload is very wide. I originally removed it (since it matched the implementation) but that caused a type error in RouterLink. So left it for now.